### PR TITLE
[apm] aggregate _dd.base_service tag for internal service overrides

### PR DESCRIPTION
**Pull request by djova**
### What does this PR do?

For internal/server service overrides (those whose `span.kind` is not `client|consumer|producer`), aggregate the `_dd.base_service` tag on APM stats in order to provide more granular visibility into the base service.

### Motivation

Related to the [Inferred Service dependencies beta](https://docs.datadoghq.com/tracing/guide/inferred-service-opt-in/?tab=java), this change enables computation of stats on service overrides scoped by the base service from which they are emitted. 

### Describe how to test/QA your changes

- Enable both `apm_config.peer_service_aggregation` and `apm_config.compute_stats_by_span_kind`.
- Send spans to the agent with `span.kind=internal` and the `service` overridden to something other than the top level `DD_SERVICE`. The tracer will set `_dd.base_service:{DD_SERVICE}` when this is done. 
- Validate that the `base_service:{DD_SERVICE}` tag is available on the service override APM metrics 

### QA Results

I emitted spans with `span.kind=internal` and confirmed that the `base_service` tag appeared correctly on the resulting APM stats. 

image

### CODE REVIEW

**Review by songy23 — APPROVED** [review]
Approval for OTel

File: pkg/trace/stats/aggregation.go [hunk 1]
```diff
@@ -99,15 +114,13 @@ func NewAggregationFromSpan(s *pb.Span, origin string, aggKey PayloadAggregation
...
-				peerTags = append(peerTags, t+":"+v)
-			}
+	for _, t := range peerTagKeysToAggregateForSpan(agg.SpanKind, s.Meta[tagBaseService], peerTagKeys) {
+		if v, ok := s.Meta[t]; ok && v != "" {
+			v = obfuscate.QuantizePeerIPAddresses(v)
+			peerTags = append(peerTags, t+":"+v)
 		}
-		agg.PeerTagsHash = peerTagsHash(peerTags)
 	}
+	agg.PeerTagsHash = peerTagsHash(peerTags)
```
Comment (ajgajg1134) [inline pkg/trace/stats/aggregation.go:123]: Any reason the PeerTagsHash is always attached now while previously we only calculated it and attached it when peerTags was non-empty?

Comment (djova) [reply pkg/trace/stats/aggregation.go:123]: > Any reason the PeerTagsHash is always attached now while previously we only calculated it and attached it when peerTags was non-empty?

Yes, it just made the code simpler. The result is still identical, where if there are no peer tags it's zero. 

```
func peerTagsHash(tags []string) uint64 {
	if len(tags) == 0 {
		return 0
	}
```

**Review by ajgajg1134 — APPROVED** [review]

File: releasenotes/notes/apm-aggregate-base-service-tag-internal-service-overrides-7c8c0a1787e0508c.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    For internal service overrides, aggregate APM stats only on `_dd.base_service`, excluding all other peer tags.
```
Comment (domalessi) [inline releasenotes/notes/apm-aggregate-base-service-tag-internal-service-overrides-7c8c0a1787e0508c.yaml:11]:
```suggestion
    APM stats for internal service overrides are now aggregated by the `_dd.base_service` tag only, enhancing visibility into specific base services.
```

**Review by domalessi — APPROVED** [review]

**Pull request merged**